### ISSUE TITLE
Autofire not working

### ISSUE THREAD

**Issue by briansrls** [visualboyadvance-m/visualboyadvance-m#731]
### Expected behavior
My intuition tells me that this feature should repeatedly press the button specified

### Actual behavior
It doesn't seem to do anything.

### Steps to reproduce the behavior
Try to configure/use Autofire in GBA/GB games.

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.):
Windows/Ubuntu

If using Linux, specify if using xorg or Wayland:
xorg

Version of code (pre-built binary version, or commit ref, or just "master"):
40741580

**Comment by briansrls** [comment]
Autohold is still working

**Comment by negativeExponent** [comment]
as far as i understand how the option works, is would mean that when autofire is enable for a button, that button becomes a turbo button when pressed. this works in Classic SMB at least.

the reason probably that you do not have some sort of reaction is due to the delay rate of the buttons. some games just do not react well on succesive button presses (this happens alot in gb/gbc games)

**Comment by briansrls** [comment]
@negativeExponent I've been trying with pokemon games, both with GBA/GB.  Can you confirm if it works at all for you?  I can't get it to work with anything.

**Comment by briansrls** [comment]
@negativeExponent I just tried with Super Mario Advance 4 - Super Mario Bros 3 on GBA but I couldn't get it to work.

**Comment by negativeExponent** [comment]
nope, works fine on my end. new vba-m compile from master, fresh or stock config with only option>input>autofire> (A,B,L,R enabled), both keyboard input and for gamepad

there's probably a setting that is messing in your end.

**Comment by briansrls** [comment]
Here is what I just did:

on Master branch:  020daad

Build debug image
About->Factory Reset
Launch Game (Some mario GBA game)
Input->Autofire->A

Nothing seems to happen.  What setting could be missing?  What game are you playing, I'll try that one.

**Comment by negativeExponent** [comment]
Super Mario Advance 4 - Super Mario Bros 3 on GBA

**Comment by briansrls** [comment]
I tried on another windows device and got the same result, are you on Windows?

**Comment by negativeExponent** [comment]
i use Arch btw

**Comment by rkitover** [comment]
I tried setting autofire A in Super Mario Advance 4 on linux and windows and it works fine for me. Mario makes continuous short jumps.

**Comment by briansrls** [comment]
I'm not sure about this one, I will try to debug it on my devices, and find the cause and update when I do.

**Comment by Ocatopuss** [comment]
The problem you're experiencing is most likely due to the default setting of VBA-M Autofire is too fast (30 hit/s I conjecture) for the game you're playing to recognize it, make it looks like the Autofire not working.

The solution is (if indeed that's what's causing it): open your vbam.ini, find “AutofireThrottle”, increase its value and save.

You can also refer to this: https://github.com/visualboyadvance-m/visualboyadvance-m/issues/1465#issue-3088486569.

**Comment by Luvatar** [comment]
Was having the same problem. 

Above poster was correct. Setting the AutofireThrottle in the ini to 2 fixed all my issues.
